include/sundials/impl/sundials_errors_impl.h

```diff
@@ -0,0 +1,176 @@
+#endif
+
+/* SUNCheckCallNoRet performs the SUNDIALS function call, and checks the
+   returned error code. If an error occured, then it will log the error, set the
+   last_err value, and call the error handler. */
+#if !defined(SUNDIALS_DISABLE_ERROR_CHECKS)
+#define SUNCheckCallNoRet(call)                                          \
+  do {                                                                   \
+    SUNErrCode sun_chk_call_err_code_ = call;                            \
+    if (SUNHintFalse(sun_chk_call_err_code_ < 0))                        \
+    {                                                                    \
+      SUNHandleErr(__LINE__, __func__, __FILE__, sun_chk_call_err_code_, \
+                   SUNCTX);                                             \
+      (void)SUNGetLastErr(SUNCTX);                                      \
+    }                                                                    \
+  }                                                                      \
+  while (0)
+#else
+#define SUNCheckCallNoRet(call) (void)call
+#endif
+
+/* Same as SUNCheckCallNoRet, but returns with the error code if an error
+ * occured. */
+#if !defined(SUNDIALS_DISABLE_ERROR_CHECKS)
+#define SUNCheckCall(call)                                               \
+  do {                                                                   \
+    SUNErrCode sun_chk_call_err_code_ = call;                            \
+    if (SUNHintFalse(sun_chk_call_err_code_ < 0))                        \
+    {                                                                    \
+      SUNHandleErr(__LINE__, __func__, __FILE__, sun_chk_call_err_code_, \
+                   SUNCTX);                                             \
+      return sun_chk_call_err_code_;                                     \
+    }                                                                    \
+  }                                                                      \
+  while (0)
+#else
+#define SUNCheckCall(call) (void)call
+#endif
+
+/* Same as SUNCheckCall, but returns with NULL. */
+#if !defined(SUNDIALS_DISABLE_ERROR_CHECKS)
+#define SUNCheckCallNull(call)                                           \
+  do {                                                                   \
+    SUNErrCode sun_chk_call_err_code_ = call;                            \
+    if (SUNHintFalse(sun_chk_call_err_code_ < 0))                        \
+    {                                                                    \
+      SUNHandleErr(__LINE__, __func__, __FILE__, sun_chk_call_err_code_, \
+                   SUNCTX);                                             \
+      return NULL;                                                       \
+    }                                                                    \
+  }                                                                      \
+  while (0)
+#else
+#define SUNCheckCallNull(call) (void)call;
```

Comment:
Should this have semicolon at end?

Reply:
I dont think so, ill remove it and see what happens!